torch/csrc/api/include/torch/optim/optimizer.h

```diff
@@ -66,12 +67,314 @@ class TORCH_API OptimizerOptions {
   virtual void set_lr(const double lr);
 };
 
-template <typename Derived>
+// Forward declarations for optimizer option types
```

Comment:
Can we make the following classes and structs private as well?

Reply:
since these are forward declarations, I'm inclined to not change the style to use the prefix (would need touching all optimizer files). I do have to change them to be struct (which resolves the inconsistency causing clang build failure).